### ISSUE TITLE
[New Data Source]: Datasource for Organizations Policies

### ISSUE THREAD

**Issue by danielw-aws** [hashicorp/terraform-provider-aws#30682]
### Description

There currently is no way to list all of the policies that are associated with a target account outside of the root account making it hard to define what has been applied to a root, OU or account. More specifically 4 different types of policies (SERVICE_CONTROL_POLICY | TAG_POLICY | BACKUP_POLICY | AISERVICES_OPT_OUT_POLICY) can all be applied to an OU, account or root but currently only the root can have policies listed. By providing this data source this allows any account in an organization to provide all of the policies associated.

### Requested Resource(s) and/or Data Source(s)

aws_organizations_organizational_policies

### Potential Terraform Configuration

```terraform
This would be a wrapping of the ListPolciesForTarget which can target a ou,root or an account to provide all of the polices that are associated with the target id.
```

### References

https://docs.aws.amazon.com/organizations/latest/APIReference/API_ListPoliciesForTarget.html

### Would you like to implement a fix?

Yes

**Comment by pshuman-heb** [comment]
Somewhat oddly, and maybe worth filing a separate bug, but the published docs for Provider 5.1.0 list a Data that purports to do this:

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/organizations_policy

With full example code that includes a misspelling of "organizational" as `oorganizational`.

**Comment by danielw-aws** [comment]
The resource and code don't exist in 5.1.0. I had tried doing two parallel pull requests for the same release one relying on the other. One got accepted the other one (this one) got punted with some fixes. The current pull request that closes this will update the documentation to be correct and not have a type and enable organizations_policies_for_target once its accepted.
